docs/design.md

```diff
@@ -0,0 +1,79 @@
+# Desing
+
+
+## Why not rules_docker
+We have reasons not to use rules_docker. While some of are performance related, others are related to its complexity. for the sake of those who does not know 
+how rules_docker work, i will keep it simple. 
+
+- it is huge in terms of written lines which makes it handle to maintain and understand if you are not familiar with it already.
```

Comment:
I'd rather remove it at this point as this does not justify why one should create yet another ruleset just because the other one is huge or hard to maintain.